client/core/types/vote.go

```diff
@@ -217,3 +218,39 @@ func (vote *Vote) String() string {
 }
 
 type Votes []*Vote
+
+type VotesSet struct {
+	m map[common.Hash]*Vote
+	l sync.RWMutex
+}
+
+func NewVotes() *VotesSet {
+	return &VotesSet{m: make(map[common.Hash]*Vote)}
+}
+
+func (v *VotesSet) Add(vote *Vote) {
+	v.l.Lock()
+	v.m[vote.Hash()] = vote
+	v.l.Unlock()
+}
+
+func (v *VotesSet) Has(h common.Hash) bool {
```

Comment:
I'd rename it to `ContainsHash`.
Similar data structures use `Contains` instead of `Has` to check if an item is in the collection.
Also adding the `Hash` part is because his is a `VoteSet`, so I'd assume a `Contains` function would receive a `Vote` to check if it exists, so if we're checking its hash I think it's clearer to name it that way.

Reply:
I do like `Contains(vote *Vote)` to hide the imlementation